### PR TITLE
get-started: initial refactoring

### LINKED ISSUE #747: docs: move /get-started index to root and /install above /get-started
(Probably after #738)

I like that **Get Started** is the first section in our docs, but ~~its 2nd page links to **Installation** for the user to go and figure out how to best install the tool. This may kind of break the continuity of get-started from the very beginning.~~ Perhaps if we have **Installation** on top in the left side bar, most users will already have DVC installed by the time they get to the Get Started -> Initialize page.

### PR DESCRIPTION
**Pull request by jorgeorpinel**
- Closes #747 (dedicated landing page for docs)
- Addresses part of #1035 (merge get-started and tutorials sections)

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
@shcheklein so far this only implements #747 but probably a little hacky – see `getItemByPath` in src/utils/sidebar.js and `SidebarMenu.timeout` in src/components/Documentation/SidebarMenu/index.js. Can you guys look at it and lmk what a better way to implement this could be? Cc @fabiosantoscode 

Also if this should be a PR of it's own I can separate it from the remaining changes (Get Started related).

**Comment by shcheklein** [comment]
Looks good to me! We should spend a bit of time to think about the landing page content (check other projects). Also, see this discussion that is relevant to your changes in the sidebar.json -  https://github.com/iterative/dvc.org/pull/1025#commitcomment-37819121 ... cc @iAdramelk - if we can merge this one before Gatsby migration we might just disallow having any entry w/o index page? But to be honest I would still do redirects if file is missing, and we should be doing this on the middleware level automatically (not relying on sidebar.json).

**Comment by shcheklein** [comment]
another point - we need to introduce some special "Home" entry to the sidebar?

**Comment by jorgeorpinel** [comment]
> - [ ] We should spend a bit of time to think about the landing page content (check other projects)

Working on that ⏳ 

> we need to introduce some special "Home" entry to the sidebar?

I don't think so, we have the DOC link in the top menu. But if you prefer that then sure, that wouldn't require the js code changes I did.

> if we can merge this one before Gatsby migration

If you want to merge this PR soon, please approve it and I'll limit it to this first step for now.

**Comment by shcheklein** [comment]
> I don't think so, we have the DOC link in the top menu. But if you prefer that then sure, that wouldn't require the js code changes I did.

yeah, I think it's good to clearly see the structure and being able to return to the page - here is an example https://www.algolia.com/doc/ .... we might even want to use an icon and separate it a bit from the other menu items, etc - need to think a bit about UX

> if we can merge this one before Gatsby migration

I would confirm with @iAdramelk first.

**Comment by jorgeorpinel** [comment]
> think about the landing page content

Updated

> I think it's good to clearly see the structure and being able to return to the page

OK, done.

**Comment by jorgeorpinel** [comment]
> I would confirm with @iAdramelk first.

Well, too late! Haha. I've merged get-started and tutorials into [/doc/learn](https://dvc-landing-refactor-ge-nouj86.herokuapp.com/doc/learn) now (so this PR has become big because of all the link renaming). What do you think?

**Comment by shcheklein** [comment]
I think it can be very hard to merge both now. I would focus on the content for now, not significant structural changes.

**Comment by shcheklein** [comment]
Or at least, we'll need to merge Gatsby migration first and then deal with this here.

**Comment by shcheklein** [comment]
Screen Shot 2020-03-15 at 8 39 13 PM


we should remove triangle, try to use icon and separate it from other items I think

**Comment by shcheklein** [comment]
https://dvc-landing-refactor-ge-nouj86.herokuapp.com/doc/Home - url should be lower case

**Comment by shcheklein** [comment]
content for the home page - probably should provide some good entry point and navigation for other sections

**Comment by shcheklein** [comment]
also, `learn` - name is too general now, what is the plan for this section?

**Comment by jorgeorpinel** [comment]
> we should remove triangle

Extracted to #1053

> learn - name is too general now, what is the plan for this section?

"Learn" is a common name for this kind of section with all the guides, tutorials, etc e.g. https://nextjs.org/learn 

---

I addressed your other couple comments @shcheklein.

**Comment by jorgeorpinel** [comment]
I'm gonna limit the scope of this PR to what's done now. It's too big already. Fortunately its mostly due to renaming /doc/tutorials -> /doc/learn

**Comment by jorgeorpinel** [comment]
> I would focus on the content for now, not significant structural changes

I thought structural changes for Get Started was the priority 😅 (both this one and then #599 which I've been working on as well). Should I postpone this work for now and focus on another high priority that is more content-related e.g. redesign the older use cases?

**Comment by shcheklein** [comment]
@jorgeorpinel 

>  I thought structural changes for Get Started was the priority

that's true, bu to my mind the biggest potion of this is content - examples, preparing repos, writing text, etc - more than moving sections around.

**Comment by jorgeorpinel** [comment]
> From https://github.com/iterative/dvc.org/issues/599#issuecomment-599883773
> ... Learn is not even part of Docs. Mostly because it's too generic. Second, it looks like the whole Learn section for Next is actually one single interactive tutorial

OK, I reverted back to /doc/tutorials but I still need to finish merging all the conflicts from the Gatsby migration... ⏳

**Comment by jorgeorpinel** [comment]
Merged. https://dvc-landing-refactor-ge-nouj86.herokuapp.com/doc is running on Gatsby now.

File: redirects-list.json [hunk 1]
```diff
@@ -6,12 +6,13 @@
   "^/((?:deb|rpm)/.+)                       https://s3-us-east-2.amazonaws.com/dvc-s3-repo/$1 303",
   "^/(?:help|chat)/?$                       https://discordapp.com/invite/dvwXA2N 303",
   "^/(?:docs|documentation)(/.*)?$          /doc$1",
-  "^/doc/commands-reference(/.*)?$          /doc/command-reference$1",
+  "^/doc/?$                                 /doc/tutorials/get-started 307",
```
Comment (shcheklein) [inline redirects-list.json]: should it be home now?

File: redirects-list.json [hunk 2]
```diff
@@ -6,12 +6,13 @@
...
+  "^/doc/get-started                        /doc/tutorials/get-started",
   "^/doc/tutorial/?$                        /doc/tutorials",
   "^/doc/tutorial/(.*)?                     /doc/tutorials/deep/$1",
+  "^/doc/commands-reference(/.*)?$          /doc/command-reference$1",
   "^/doc/use-cases/data-and-model-files-versioning/?$ /doc/use-cases/versioning-data-and-model-files",
-  "^/doc/?$                                 /doc/get-started 307",
+  "^/doc/user-guide/contributing/?$         /doc/user-guide/contributing/core 307",
   "^/doc/understanding-dvc/?$               /doc/understanding-dvc/collaboration-issues 307",
-  "^/doc/changelog/?$                       /doc/changelog/0.18 307",
```
Comment (shcheklein) [inline redirects-list.json]: I really want to get rid of changelog, make it an external link ... again a special case like a Home button ... we spend time on it again and again

File: src/components/Community/data.json [hunk 3]
```diff
@@ -28,13 +28,13 @@
...
     {
-      "url": "/doc/get-started",
+      "url": "/doc/tutorials/get-started",
       "title": "Get started",
       "description": "Reproducible walkthroughs on advanced topics"
     },
     {
       "url": "/doc/tutorials",
-      "title": "Tutorials",
+      "title": "DVC Tutorials",
```
Comment (shcheklein) [inline src/components/Community/data.json]: why DVC Tutorials?

File: redirects-list.json (hunk 1 above)
Comment (jorgeorpinel) [reply redirects-list.json]: Yes, I also noticed this earlier, it's fixed. Thanks

File: redirects-list.json (hunk 2 above)
Comment (jorgeorpinel) [reply redirects-list.json]: I thought it was an external link already before? I didn't introduce this in the Gatsby migration, just moved the order of some redirects.

File: src/components/Community/data.json (hunk 3 above)
Comment (jorgeorpinel) [reply src/components/Community/data.json]: I was changing just back from "Learn DVC", didn't remember it was just "Tutorials" before. Reverted.

**Comment by jorgeorpinel** [comment]
Let's try to address the remaining refactoring in #1074 (or further PRs)? Any problems specific to these nav order changes I can still fix here, of course.

**Pull request merged**